### PR TITLE
UI: Add basic HUD to the game

### PR DESCRIPTION
**Pull request by Dennisbonke**
This PR adds a basic player HUD to the game, comes with current action points and an image of the player that you are controlling. Health bar is being a giant pain so I haven't done that one yet.

### CODE REVIEW

File: Assets/Scripts/UI/HUD/ActivePlayer.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+using UnityEngine;
+using UnityEngine.UI;
+using OperationBlackwell.Core;
+
+namespace OperationBlackwell.UI {
+	public class ActivePlayer : MonoBehaviour {
+		[SerializeField] private Image activePlayer_;
+
+		private void Update() {
```
Comment (Westlanderz) [inline Assets/Scripts/UI/HUD/ActivePlayer.cs:9]: Please do this with events, the Update has a lot of overhead if the player does not change that frame.

File: Assets/Scripts/UI/HUDController.cs [hunk 2]
```diff
@@ -21,14 +22,25 @@ public class HUDController : MonoBehaviour {
...
-				actionPointsText_.text = "";
+				actionPointsText_.text = "0";
 			}
 		}
 
 		private void UpdateTurnCounter(object sender, int turn) {
 			turnCounterText_.text = "Turn: " + turn.ToString();
 		}
+
+		private void Update() {
```
Comment (Westlanderz) [inline Assets/Scripts/UI/HUDController.cs:36]: Cant this be done on an event either?

**Review by Westlanderz — CHANGES_REQUESTED** [review]
Looks good in general, just 2 comments about doing updates in the `Update()` function.

File: Assets/Scripts/Player/UnitGridCombat.cs [hunk 3]
```diff
@@ -429,5 +430,9 @@ private enum Direction {
 			animator_.SetBool("isShootingLeft", false);
 			animator_.SetBool("isShootingRight", false);
 		}
+
+		public override string GetName() {
```
Comment (Westlanderz) [inline Assets/Scripts/Player/UnitGridCombat.cs:434]: Why do you use `string` here while in both the name variable and in the original function declaration you use `String`?

File: Assets/Scripts/UI/HUD/ActivePlayer.cs (hunk 1 above)
Comment (Dennisbonke) [reply Assets/Scripts/UI/HUD/ActivePlayer.cs]: Done

File: Assets/Scripts/UI/HUDController.cs (hunk 2 above)
Comment (Dennisbonke) [reply Assets/Scripts/UI/HUDController.cs]: Done

**Review by Westlanderz — CHANGES_REQUESTED** [review]
It all looks good except for the one nitpick comment.

**Review by Westlanderz — APPROVED** [review]
LGTM

**Pull request merged**